src/main.rs

```diff
@@ -146,6 +160,240 @@ fn display_sysinfo() {
     );
 }
 
+#[derive(Serialize, Deserialize)]
+struct DecodedData {
+    bytecode: Vec<ELFInstruction>,
+    memory_init: Vec<(u64, u8)>,
+}
+
+fn preprocess_and_save(func_name: &str) -> Result<()> {
+    let mut program = Program::new("guest");
+    program.set_func(func_name);
+    program.set_std(false);
```

Comment:
Can we figure out how to extract these options from the macro? This is how we currently do it for normal usage from the macro generated functions, which I am sure we could extract if needed.

Reply:
How would it be if we also read out whether the feature is set or not in the preprocessing, in which we modify Cargo.toml anyway?

Reply:
Also, for the todo comment below I thought about extracting the parse_attributes function,  for example into "common". This way we could reuse it both in preprocessing and in the macro. What do you think of this idea?

Reply:
Yeah I think that is fine. It's just important that when we use this preprocess function it behaves the same way as it would in the non-wasm variants so that the prover/verifier are interoperable. Otherwise a proof generated would set the various attributes here but the wasm verifier would have these defaults instead.